cylc/flow/network/server.py

```diff
@@ -103,10 +106,32 @@ def start(self, min_port, max_port):
             min_port (int): minimum socket port number
             max_port (int): maximum socket port number
         """
+        # create & configure an authenticator for the ZMQ context
+        self.curve_auth = ThreadAuthenticator(self.context)
+        self.curve_auth.start()  # start the authentication thread
+
+        # check for, & create if nonexistent, user keys in the right location
+        if not ensure_user_keys_exist():
+            raise CylcError("Unable to generate user authentication keys.")
+
+        # Setting the location means that the CurveZMQ auth will only accept
+        # public client certificates from the given directory, as generated by
+        # a user when they initiate a ZMQ socket ready to connect to a server.
+        self.curve_auth.configure_curve(
+            domain='*', location=UserFiles.get_user_certificate_full_path())
+
         # create socket
         self.socket = self.context.socket(zmq.REP)
         self.socket.RCVTIMEO = int(self.RECV_TIMEOUT) * 1000
 
+        # fetch server keys, generated at sutie registration, for auth
```

Comment:
s/sutie/suite

Reply:
Thanks @kinow, very good spotting on the typos. I'm on leave tomorrow (UK-time) but on Thursday I will correct these along with addressing any other feedback raised.

Please feel free to review further if you'd like to! I've assigned David since he is very familiar with the networking code by now from his own work, & then Oliver & Tim because I was asked to by my manager as we are trying to review this quickly for David, but the more eyes on the code the better. 😃 

I'm glad you found the comments helpful, that's always a good sign that a comment is doing the right job!